pyspecdata/core.py

```diff
@@ -7312,8 +7331,8 @@ def fit(self,set_what = None, set_to = None, force_analytical = False, silent =
             if not silent: print('{\\bf Warning:} You have no error associated with your plot, and I want to flag this for now\n\n')
             warnings.warn('You have no error associated with your plot, and I want to flag this for now',Warning)
             sigma = ones(shape(y))
-        if set_what != None:
-            p_ini = [1.0]*self.number_of_parameters
+        if set_what is not None:
+            p_ini = self.guess()
```

Comment:
seems more roundabout, but I'm doing this so that `guess()` can do other things, as well